proto/v1/structure/state.proto

```diff
@@ -0,0 +1,881 @@
+  oneof interaction_type {
+
+    // The interaction that takes the form of a simple Continue button.
+    ContinueInstance continue = 1;
+
+    // The interaction for fraction input.
+    FractionInputInstance fraction_input = 2;
+
+    // The interaction for item selection input.
+    ItemSelectionInputInstance item_selection_input = 3;
+
+    // The interaction for multiple item selection input.
+    MultipleChoiceInputInstance multiple_choice_input = 4;
+
+    // The interaction for number input.
+    NumericInputInstance numeric_input = 5;
+
+    // The interaction for text input.
+    TextInputInstance text_input = 6;
+
+    // The interaction for drag and drop sorting.
+    DragAndDropSortInputInstance drag_and_drop_sort_input = 7;
+
+    // The interaction allowing multiple-choice selection on an image.
+    ImageClickInputInstance image_click_input = 8;
+
+    // The interaction for ration input.
+    RatioExpressionInputInstance ratio_expression_input = 9;
+
+    // The interaction that allows the exploration to end.
+    EndExplorationInstance end_exploration = 10;
+  }
+}
+
+// Answer type: none (N/A since no answers are submitted for this interaction).
+// Structure of the continue instance.
+message ContinueInstance {
+  CustomizationArgs customization_args = 1;
+
+  // The default outcome of the continue instance.
+  Outcome default_outcome = 2;
+
+  // Continue interactions cannot have custom answer groups, and do not support
+  // solutions. Users cannot receive hints since only one answer is possible
+  // (clicking the button).
+
+  // Structure of the customization args of continue instance.
+  message CustomizationArgs {
+    // Note that while the continue interaction has a default button text
+    // defined, it's not used in the app since the default string needs to come
+    // through the app's strings so that it can be translated.
+    SubtitledHtml button_text = 1;
+  }
+}
+
+// Answer type: Fraction.
+// Structure of the fraction input interaction.
+message FractionInputInstance {
+
+  // A set of arguments to customize the fraction input.
+  CustomizationArgs customization_args = 1;
+
+  // A list of answer groups of the fraction input.
+  repeated AnswerGroup answer_groups = 2;
+
+  // The default outcome of the fraction input.
+  Outcome default_outcome = 3;
+
+  // A list of hints in the fraction input.
+  repeated Hint hints = 4;
+
+  // The solution of the fraction input.
+  Solution solution = 5;
+
+  // Structure of the customization args of the fraction input.
+  message CustomizationArgs {
+
+    // The fraction is in simplest form or not.
+    bool requires_simplest_form = 1;
+
+    // The fraction allow to be in improper fraction.
+    bool allow_improper_fractions = 2;
+
+    // The fraction contains non zero integer part or not.
+    bool allow_nonzero_integer_part = 3;
+
+    // The custom placeholders for fraction input.
+    SubtitledHtml custom_placeholder = 4;
```

Comment:
As per the current demo explorations, this is not the HTML-type data for fraction interaction. It's a `SubtitledUnicode`

```
        customPlaceholder:
          value:
            content_id: ca_customPlaceholder_2
            unicode_str: ''
```